core/src/trezor/ui/model/button/t1.py

```diff
@@ -0,0 +1,175 @@
+from trezor import ui
+from trezor.ui import display, in_area
+
+from ..types import ButtonStyle, ButtonStyleState
+
+if False:
+    from typing import Optional
+    from ..types import (  # noqa: F401
+        ButtonStyleStateType,
+        ButtonStyleType,
+        ButtonContent,
+    )
+
+
+class ButtonDefault(ButtonStyle):
+    class normal(ButtonStyleState):
+        bg_color = ui.FG
+        fg_color = ui.BG
+        text_style = ui.BOLD
+        border_color = ui.FG
+        radius = 1
+
+    class active(normal):
+        bg_color = ui.BG
+        fg_color = ui.FG
+        text_style = ui.BOLD
+        border_color = ui.BG
+        radius = 1
+
+    class disabled(normal):
+        bg_color = ui.FG
+        fg_color = ui.BG
+        text_style = ui.NORMAL
+        border_color = ui.FG
+        radius = 1
```

Comment:
A general note: these style definitions seem to be at the wrong level of abstraction here.

I'm wondering if `Button` should actually be exposed as a widget the way it is now -- as opposed to an implementation detail of, e.g., the `Confirm` widget. Then we could have wildly different implementations of `Button` for T1 and TT -- which seems to be _what we need_ -- because the button implementation would not be visible from outside the `ui.model.<model>` namespace.

I.e., in `trezor/ui/confirm.py`:
```
model = import_model_specifics(__name__)

model.show_confirm("Are you sure?", CONFIRM_TYPE_OKCANCEL)
```
in `trezor/ui/model/t1/confirm.py`
```
from . import button

button.render_left("OK")
button.render_right("Cancel")
```
in `trezor/ui/model/tt/confirm.py`
```
from . import button

button.render("OK", ButtonStyle.ButtonConfirm, GRID_LEFT)
button.render("Cancel", ButtonStyle.ButtonCancel, GRID_RIGHT)
```

...which would necessitate flipping again? I don't remember the original layout, but I think it was `ui.model.<model>.<widget>`, and I proposed `ui.model.<widget>.<model>`? If that's so, then your original version was perhaps the smarter choice.

---

not saying that this needs to happen in this PR, but it seems to be the right general direction?

Reply:
You're right, the button handling is ugly and the typing problems are a symptom. Let's do one(+) more iteration(s) and get this right before merging.

Reply:
I tried to implement this approach in #1239, PTAL. It's less polished than this PR but hopefully should be useful for comparison (typechecks and passes TT tests though).